### PR TITLE
Drop .self pattern from many constructors

### PR DESCRIPTION
**Pull request by jcorbin**
r @Raynos @kriskowal

### CODE REVIEW

**Comment by Raynos** [comment]
Yep. #legit. story checks out.

File: connection.js [hunk 1]
```diff
@@ -40,56 +40,57 @@ function TChannelConnection(channel, socket, direction, socketRemoteAddr) {
         'refusing to create self connection'
     );
 
-    var self = this;
```
Comment (kriskowal) [inline connection.js]: This occurrence appears to still be needed for callbacks below.

File: connection.js [hunk 2]
```diff
@@ -40,56 +40,57 @@ function TChannelConnection(channel, socket, direction, socketRemoteAddr) {
...
     }
 
-    self.socket = socket;
-    self.ephemeral = false;
-    self.initHeaders = null;
+    this.socket = socket;
+    this.ephemeral = false;
+    this.initHeaders = null;
 
+    var self = this;
```
Comment (kriskowal) [inline connection.js]: Found it. nm.

**Comment by kriskowal** [comment]
roger :+1: ebert :+1:

**Pull request merged**